File: functions/notes.py
# Josephine Chiu, Function Notes for Python

#function is an action stored in a key word

number = int(input("Can I get anumber\n"))

def add():
    numOne = int(input("Give me a number:\n"))
    numTwo = int(input("Give me a number:\n"))
    print(numOne+numTwo)
add() # if you put add() six times, 16 will print six times10
add()
add()

def add(numOne, numTwo): #parameters go in the parentheiss seperated by commas

    print(numOne+numTwo)

number = int(input("Can I get anumber\n"))
add(int(input("Tell me a number\n")), number) #arguments are given when the function is called AND must match the number of parameters

add(2,4)

add(7,21)

#strings
def user(word):
    return input(f"Tell me a {word}:\n")

name = user("name")
verb = user("verb")
place = user("place")

print(f"{name} was {verb} and somehow got to {place}. \n")

File: functions/notes.c
// Josephine Chiu, Functions Notes for C
#include <stdio.h>

int add(int numOne, int numTwo){
    printf("%d\n", numOne+numTwo);
    return 0;
}
const char* word(char type[20]){ //constant variable and the star means all
    char answer[50];
    printf("Please give me a %s:\n", type);
    getStr(answer, sizeof(answer)-1);
    scanf("%s", answer);
    return answer;
}

int main(void){
    //printf("hello world\n");
    //add(4, 9);
    //add(20, 2);
    //add(5, 9);
    //add(3, 6);
    //input("name");
    //nput("verb");
    //input("place");

//don't return strings in C!!!
    return 0;
}